R/zchunk_L124.nonco2_unmgd_R_S_T_Y.R

```diff
@@ -43,80 +41,182 @@ module_emissions_L124.nonco2_unmgd_R_S_T_Y_DISABLED <- function(command, ...) {
+      left_join(filter(EDGAR_history, sector == "grassland"), by = c("GCAM_region_ID", "year")) %>%         # Map in EDGAR grassland emissions
+      mutate(value = value * land_share) %>%                                                                # Compute emissions by GLU using EDGAR totals and land shares
+      na.omit() %>%
+      select(-sector, -land_share) ->
+      L124.nonco2_tg_R_grass_Y_GLU
+
+    # Part 2: Forest fires and deforestation
+    # Calculate share of forest emissions from forest fires versus deforestation using GFED data.
+    # Bind all GFED data together, aggregate by GCAM region/gas/year, calculate share of forest fire versus deforestation
+    bind_rows(mutate(GFED_ForestFire_CO, Non.CO2 = "CO", type = "ForestFire"),
+              mutate(GFED_Deforest_CO, Non.CO2 = "CO", type = "Deforest"),
+              mutate(GFED_ForestFire_SO2, Non.CO2 = "SO2", type = "ForestFire"),
+              mutate(GFED_Deforest_SO2, Non.CO2 = "SO2", type = "Deforest"),
+              mutate(GFED_ForestFire_NOx, Non.CO2 = "NOx", type = "ForestFire"),
+              mutate(GFED_Deforest_NOx, Non.CO2 = "NOx", type = "Deforest")) %>%
+      # NMVOC is split into lots of inventories, so using CO for now.
+      bind_rows(mutate(GFED_ForestFire_CO, Non.CO2 = "NMVOC", type = "ForestFire"),
+                mutate(GFED_Deforest_CO, Non.CO2 = "NMVOC", type = "Deforest")) %>%
+      # Use CO for other missing gases. Previous code did this implicitly in lines 162-166
+      bind_rows(mutate(GFED_ForestFire_CO, Non.CO2 = "CH4", type = "ForestFire"),
+                mutate(GFED_Deforest_CO, Non.CO2 = "CH4", type = "Deforest"),
+                mutate(GFED_ForestFire_CO, Non.CO2 = "N2O", type = "ForestFire"),
+                mutate(GFED_Deforest_CO, Non.CO2 = "N2O", type = "Deforest")) %>%
+      gather(year, value, -Country, -Non.CO2, -type) %>%                                             # Convert from wide to long
+      mutate(year = as.integer(year)) %>%
+      spread(type, value) %>%                                                                        # Spread data so deforestation and forest fires are in columns
+      mutate(iso = tolower( Country ), Country = NULL) %>%
+      change_iso_code('rou', 'rom') %>%                                                              # Convert Romania iso code to pre-2002 value
+      left_join(iso_GCAM_regID, by = "iso") %>%
+      group_by(GCAM_region_ID, Non.CO2, year ) %>%
+      summarize(ForestFire = sum(ForestFire), Deforest = sum(Deforest) ) %>%                         # Aggregate emissions by region, gas, and year
+      na.omit() %>%
+      mutate(PctForestFire = ForestFire / (ForestFire + Deforest)) %>%                               # Compute share of emissions from forest fires
+      mutate(PctForestFire = if_else( is.na(PctForestFire), 1, PctForestFire )) %>%                  # Assume missing values mean 100% forest fires since these are easier to model in GCAM
```

Comment:
We have NAs in the GCAM_region_ID prior to the na.omit because there are iso codes in the GFED data that don't exist in the iso_to_region mapping ("ala" "ata" "atf" "blm" "bvt" "ggy" "hmd" "iot" "jey" "maf" "sgs" "umi"). We have NAs after that mutate statement because there are places where the denominator (`ForestFire + Deforest`) is zero. I'll add comments to make this clearer.